src/core/public/overlays/sidecar/sidecar_service.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/* eslint-disable max-classes-per-file */
+
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { BehaviorSubject, Observable, ReplaySubject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { I18nStart } from '../../i18n';
+import { MountPoint } from '../../types';
+import { OverlayRef } from '../types';
+import { Sidecar } from './components/sidecar';
+/**
+ * A SidecarRef is a reference to an opened sidecar panel. It offers methods to
+ * close the sidecar panel again. If you open a sidecar panel you should make
+ * sure you call `close()` when it should be closed.
+ * Since a sidecar could also be closed by a user or from another sidecar being
+ * opened, you must bind to the `onClose` Promise on the Sidecar instance.
+ * The Promise will resolve whenever the sidecar was closed at which point you
+ * should discard the SidecarRef.
+ *
+ * @public
+ */
+class SidecarRef implements OverlayRef {
+  /**
+   * An Promise that will resolve once this sidecar is closed.
+   *
+   * Sidecars can close from user interaction, calling `close()` on the sidecar
+   * reference or another call to `open()` replacing your sidecar.
+   */
+  public readonly onClose: Promise<void>;
+
+  private closeSubject = new Subject<void>();
+
+  constructor() {
+    this.onClose = this.closeSubject.toPromise();
+  }
+
+  /**
+   * Closes the referenced sidecar if it's still open which in turn will
+   * resolve the `onClose` Promise. If the sidecar had already been
+   * closed this method does nothing.
+   */
+  public close(): Promise<void> {
+    if (!this.closeSubject.closed) {
+      this.closeSubject.next();
+      this.closeSubject.complete();
+    }
+    return this.onClose;
+  }
+}
+
+/**
+ * APIs to open and manage sidecar.
+ *
+ * @public
+ */
+export interface OverlaySidecarStart {
+  /**
+   * Opens a sidecar panel with the given mount point inside. You can use
+   * `close()` on the returned SidecarRef to close the sidecar.
+   *
+   * @param mount {@link MountPoint} - Mounts the children inside a sidecar panel
+   * @param options {@link OverlaySidecarOpenOptions} - options for the sidecar
+   * @return {@link SidecarRef} A reference to the opened sidecar panel.
+   */
+  open(mount: MountPoint, options: OverlaySidecarOpenOptions): SidecarRef;
+
+  /**
+   * Override the default support config
+   * @param config The updated support config
+   */
+  setSidecarConfig(config: Partial<ISidecarConfig>): void;
+
+  /**
+   * Get an observable of the current locked state of the nav drawer.
+   */
+  getSidecarConfig$(): Observable<ISidecarConfig | undefined>;
+
+  /**
+   * Hide side car.
+   */
+  hide: () => void;
+
+  /**
+   * Show side car.
+   */
+  show: () => void;
```

Comment:
A little curious why these methods are not in SidecarRef, they seem like instance's methods.